### PR TITLE
#47: Added rank column

### LINKED ISSUE #47: Add rank column to model results returning top k predictions
- FillingMask and QuestionAswering models might return top_k results
- We better to add rank column to results to make selection easier.
    - Rank the prediction with k results.

### PR DESCRIPTION
**Pull request by umitbuyuksahin**
Fix #47

### CODE REVIEW

File: tests/integration_tests/with_db/udfs/test_filling_mask_script.py [hunk 1]
```diff
@@ -35,5 +35,5 @@ def test_filling_mask_script(
 
     # assertions
     n_rows_result = n_rows * top_k
-    n_cols_result = len(input_data[0]) + 1  # + 2 new cols -1 device_id col
+    n_cols_result = len(input_data[0]) + 2  # + 3 new cols -1 device_id col
```
Comment (Nicoretti) [inline tests/integration_tests/with_db/udfs/test_filling_mask_script.py:38]:
```suggestion
    ADDED_COLUMNS = 3
    REMOVED_COLUMNS = 1 # device_id
    n_cols_result = len(input_data[0]) + (ADDED_COLUMNS - REMOVED_COLUMNS) 
```

File: tests/integration_tests/without_db/udfs/test_filling_mask_udf.py [hunk 2]
```diff
@@ -85,6 +85,13 @@ def test_filling_mask_udf(
...
-           list(result_df.columns) == columns[1:] + ['filled_text', 'score']
+    new_columns = ['filled_text', 'score', 'rank']
+    assert result_df.shape == (n_rows * top_k, len(columns)+len(new_columns)-1) \
+           and result_df.shape[1] == len(columns) + len(new_columns) - 1 \
+           and list(result_df.columns) == columns[1:] + new_columns \
+           and result_df['score'].dtypes == 'float' \
+           and result_df['rank'].dtypes == 'int' \
+           and all([result_df[row*top_k: top_k + row*top_k]
+                   .sort_values(by='score', ascending=False)['rank']
+                   .is_monotonic for row in range(n_rows)])
```
Comment (Nicoretti) [inline tests/integration_tests/without_db/udfs/test_filling_mask_udf.py:96]: The assertions are very hard to ready from my point of view, what about doing something like this?

```python
    has_valid_shape = result_df.shape == (n_rows * top_k, len(columns)+len(new_columns)-1)
    score_is_typed_as_float = result_df['score'].dtypes == 'float'
    rank_is_typed_as_int = and result_df['rank'].dtypes == 'int'
    ...

    assert all((
        has_valid_shape, 
        score_is_typed_as_float,
        rank_is_typed_as_int,
    ))

```

File: tests/integration_tests/without_db/udfs/test_question_answering_udf.py [hunk 3]
```diff
@@ -92,7 +92,11 @@ def test_question_answering_udf(
     sequence_classifier.run(ctx)
 
     result_df = ctx.get_emitted()[0][0]
-    new_columns = ['answer', 'score']
+    new_columns = ['answer', 'score', 'rank']
     assert result_df.shape[1] == len(columns) + len(new_columns) - 1 \
            and list(result_df.columns) == columns[1:] + new_columns \
-           and result_df['score'].dtypes == 'float'
+           and result_df['score'].dtypes == 'float' \
```
Comment (Nicoretti) [inline tests/integration_tests/without_db/udfs/test_question_answering_udf.py:98]: see suggestion regarding the asserts in `test_filling_mask_udf.py`

File: tests/integration_tests/without_db/udfs/test_filling_mask_udf.py (hunk 2 above)
Comment (umitbuyuksahin) [reply tests/integration_tests/without_db/udfs/test_filling_mask_udf.py:96]: thanks, agree, this way is better for readability

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**